tests/models/user_preference_model_tests.py

```diff
@@ -0,0 +1,61 @@
+"""
+ARCHES - a program developed to inventory and manage immovable cultural heritage.
+Copyright (C) 2013 J. Paul Getty Trust and World Monuments Fund
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import random
+import os, uuid
+from django.test import TransactionTestCase
+from django.test.utils import captured_stdout
+from django.db import connection, connections
+from django.core import management
+from tests.base_test import ArchesTestCase
+from arches.app.models import models
+from arches.app.models.models import UserPreference
+from arches.app.utils.data_management.resources.importer import BusinessDataImporter
+from tests import test_settings
+
+# these tests can be run from the command line via
+# python manage.py test tests.models.user_preference_model_tests --settings="tests.test_settings"
+
+
+class UserPreferenceTests(ArchesTestCase):
+    def generate_user_preference(self):
```

Comment:
Test data reused in multiple tests is usually created in `setUpTestData()` so that it runs once per class.
```suggestion
    @classmethod
    def setUpTestData(cls):
        cls.user_preference = UserPreference.objects.create(
            ...
            
```

Reply:
But see below, we may not need this module unless it's a placeholder for future work.

Reply:
I have added a setUpTestData() function in the API tests, as it makes sense to have some test cases in the db for testing GET and DELETE.  I couldn't fit this function in the model tests, as they are testing saving new UserPreferences with potential issues.